operations/mimir/querier.libsonnet

```diff
@@ -28,6 +28,9 @@
       // We request high memory but the Go heap is typically very low (< 100MB) and this causes
       // the GC to trigger continuously. Setting a ballast of 256MB reduces GC.
       'mem-ballast-size-bytes': 1 << 28,  // 256M
+
+      // Don't depend on Memcached on startup.
+      'blocks-storage.bucket-store.metadata-cache.memcached.dns-ignore-startup-failures': true,
```

Comment:
Should we make this the default value for the CLI flag, rather than setting it in Jsonnet and Helm?

(same comment applies below)

Reply:
Changing the defaults is probably fine, when I think about it. It does make things simpler.

Reply:
I realized it's not that simple after all @charleskorn - `-dns-ignore-startup-failures` is a dskit flag. dskit being an open source project, I don't think we can just go and change its flag defaults. WDYT?

Reply:
> I realized it's not that simple after all @charleskorn - -dns-ignore-startup-failures

It's experimental and added by @tinitiuset specifically for Mimir so that's fine if we want to do it.

If you're going to add the flag to Mimir, can you move this to `config.libsonnet` with all the other cache settings?

Reply:
@56quarters Thanks, in that case it shouldn't be a problem to upstream the new default to dkskit. I'll do it.

Reply:
@56quarters @charleskorn [dskit PR](https://github.com/grafana/dskit/pull/705).

Reply:
Thanks for the review @charleskorn! The dskit PR is merged.